tensorflow/python/ops/custom_gradient.py

```diff
@@ -165,6 +165,73 @@ def second_order_and_transpose(ddy):
   Additional arguments to the inner `@tf.custom_gradient`-decorated function
   control the expected return values of the innermost function.
 
+  The examples above illustrate how to specify custom gradients for functions
+  which do not read from variables. The following example uses variables, which
+  require special handling because they are effectively inputs of the forward
+  function.
+
+  ```python
+  >>> weights = tf.Variable(tf.ones([2]))  # Trainable variable weights
+  >>> @tf.custom_gradient
+  ... def linear_poly(x):
+  ...   # Creating polynomial
+  ...   poly = weights[1] * x + weights[0]
+  ...
+  ...   def grad_fn(dpoly, variables=None):
+  ...     # dy/dx = weights[1] and we need to left multiply dpoly
+  ...     grad_xs = dpoly * weights[1]  # Scalar gradient
+  ...
+  ...     input_shape = len(x.shape)  # x contains how many dimensions
+  ...     grad_vars = []  # To store gradients of passed variables
+  ...     if variables is not None:
+  ...       for v in variables:
+  ...         if v is weights:
+  ...           # Manually computing dy/dweights
+  ...           dy_dw = tf.stack([x ** 1, x ** 0])
+  ...           if input_shape > 0:
+  ...             # Note here that reduce operation is performed
+  ...             # in axis so that it computes gradient of specific
+  ...             # example in batch and it doesn't reduce the entire batch
+  ...             # that is why we are only reducing on axis 1.
```

Comment:
Maybe to simplify the example just omit the batching bits and the `else:` and `tf.reduce_sum(tf.reshape(dy_dw, [2, -1]), axis=1)` instead

Reply:
done